### PR TITLE
docs(lb4): customizing sequence actions

### LINKED ISSUE #863: [Docs]: How to create and register a custom sequence action (e.g. send)
## Overview

We have removed "Configurable response types/formats #436" from MVP because custom responses can be produced via a custom send action. However, our documentation does not explain how to implement such custom send actions or how to register them in an application.

We need to do better! 😄 

See https://github.com/strongloop/loopback-next/issues/436#issuecomment-349073924, https://github.com/strongloop/loopback-next/issues/436#issuecomment-349329054 and https://github.com/strongloop/loopback-next/issues/436#issuecomment-349387123. I am cross-posting the content of the last comment below.

## Acceptance Criteria
- [x] Take the snippet below and convert it into documentation describing the creation of custom sequences
- [x] Update loopback.io with this documentation (http://loopback.io/doc/en/lb4/Sequence.html#using-sequence-actions)
---

Registering your custom send action involves injecting on the `RestBindings.SequenceActions.SEND` key with a custom Provider for Send.

**application.ts**
```ts
export class YourApp extends RepositoryMixin(Application) {
  constructor() {
    super();
    // Assume your controller setup and other items are in here as well.
    this.bind(RestBindings.SequenceActions.SEND).toProvider(CustomSendProvider);
  }
```

**custom-send-provider.ts**
```ts
import { Send } from "@loopback/rest";
import { Provider, BoundValue } from "@loopback/context";
import { writeResultToResponse } from "@loopback/rest";

export class CustomSendProvider implements Provider<BoundValue>{
  value(): Send | Promise<Send> {
    return writeResultToResponse; // Replace this with your own implementation.
  }
}
```

If you're implementing _multiple servers_ in your application, binding this key at the Application-level context will make that binding the default for all of the servers connected to it. This means that if you wanted different custom providers for different RestServer instances, you'd want to perform those bindings elsewhere (like in an async override of the `start` method).

In this example, PrivateSendProvider might allow stack traces to be returned via the response, whereas PublicSendProvider will not.

**in application.ts**
```ts
async start() {
   const publicServer = await this.getServer('public');
   const privateServer = await this.getServer('private');
   publicServer.bind(RestBindings.SequenceActions.SEND).toProvider(PublicSendProvider);
   privateServer.bind(RestBindings.SequenceActions.SEND).toProvider(PrivateSendProvider);
   await super.start(); // Don't forget to call start!
}
```

### PR DESCRIPTION
**Pull request by kjdelisle**
closes https://github.com/strongloop/loopback-next/issues/863

- Change the current section to specify that we're customizing the sequence actions
- Add a brief explanation of what we're doing
- Show sample code
- Brief conclusion about what it lets you do, as well as mentioning that this works for all listed sequence actions

### CODE REVIEW

File: pages/en/lb4/Sequence.md [hunk 1]
```diff
@@ -141,11 +141,58 @@ class MySequence extends DefaultSequence {
...
+  }
+}
+```
+
+Next, in our application class, we'll inject this provider on the
+`RestBindings.SequenceActions.SEND` key.
+
+{% include code-caption.html content="/src/application.ts" %}
+```ts
+export class YourApp extends RepositoryMixin(Application) {
```
Comment (b-admike) [inline pages/en/lb4/Sequence.md:183]: I suggest importing `RepositoriyMixin` and `Application` here to be consistent with the snippet above.

Comment (jannyHou) [reply pages/en/lb4/Sequence.md:183]: And import `CustomSendProvider` as well :-p

File: pages/en/lb4/Sequence.md [hunk 2]
```diff
@@ -141,11 +141,58 @@ class MySequence extends DefaultSequence {
...
+{% include code-caption.html content="/src/providers/custom-send-provider.ts" %}
+**custom-send-provider.ts**
+```ts
+import { Send } from "@loopback/rest";
+import { Provider, BoundValue } from "@loopback/context";
+import { writeResultToResponse } from "@loopback/rest";
+
+export class CustomSendProvider implements Provider<BoundValue>{
+  value(): Send | Promise<Send> {
+    return writeResultToResponse; // Replace this with your own implementation.
```
Comment (bajtos) [inline pages/en/lb4/Sequence.md:173]: I am proposing to enhance this code snippet to show our users how to best structure their code.

```ts
export class CustomSendProvider implements Provider<BoundValue>{
  constructor(
    // specify dependencies to inject 
  ) {}

  value(): Send | Promise<Send> {
    return (response, result) => this.action(response, result);
  }

  action(response: Response, result: OperationRetval): void {
    // put your implementation here
    // access dependencies injected to the constructor via "this" properties
  }
}
```

**Review by bajtos — APPROVED** [review]
👏

**Comment by bajtos** [comment]
strongloop/loopback-next#863 is mentioning configuration of custom sequence actions in multi-server setup. I know multiple servers are out of scope of MVP, but I think our documentation should at least briefly mention what will happen in such case (all servers will use the same customization) and that instructions for customizing sequence actions per-server instance will be provided later.

File: pages/en/lb4/Sequence.md (hunk 1 above)
Comment (bajtos) [inline pages/en/lb4/Sequence.md:183]: Should we extend `RestApplication` instead of `Application`?

**Review by shimks — APPROVED** [review]
LGTM once you address @bajtos 's comments

**Review by bschrammIBM — APPROVED** [review]

**Comment by kjdelisle** [comment]
@bschrammIBM I needed to update this pull request with a few changes based on @bajtos 's recommendations. I'll open a follow-up PR to address that feedback instead.

**Pull request merged**